sw/host/opentitanlib/src/test_utils/rpc.rs

```diff
@@ -58,19 +59,29 @@ impl<T: DeserializeOwned> UartRecv for T {
                 let cap = console
                     .captures(ExitStatus::ExitSuccess)
                     .expect("RESP_OK capture");
-                let s = cap.get(1).expect("RESP_OK group").as_str();
-                Ok(serde_json::from_str::<Self>(s)?)
+                let json_str = cap.get(1).expect("RESP_OK group").as_str();
+                let crc_str = cap.get(2).expect("CRC not provided").as_str();
+                check_crc(json_str, crc_str);
+                Ok(serde_json::from_str::<Self>(json_str)?)
             }
             ExitStatus::ExitFailure => {
                 let cap = console
                     .captures(ExitStatus::ExitFailure)
                     .expect("RESP_ERR capture");
-                let s = cap.get(1).expect("RESP_ERR group").as_str();
-                let err = serde_json::from_str::<Status>(s)?;
+                let json_str = cap.get(1).expect("RESP_ERR group").as_str();
+                let crc_str = cap.get(2).expect("CRC not provided").as_str();
+                check_crc(json_str, crc_str);
+                let err = serde_json::from_str::<Status>(json_str)?;
                 Err(err.into())
             }
             ExitStatus::Timeout => Err(UartError::GenericError("Timed Out".into()).into()),
             _ => Err(anyhow!("Impossible result: {:?}", result)),
         }
     }
 }
+
+fn check_crc(json_str: &str, crc_str: &str) {
+    let crc = u32::from_str_radix(crc_str, 16).expect("Couldn't parse CRC string.");
+    let actual_crc = Crc::<u32>::new(&CRC_32_ISO_HDLC).checksum(json_str.as_bytes());
+    assert!(crc == actual_crc);
```

Comment:
Should this function return a `Result<()>` instead of panicking when the CRC is wrong?